frontend/src/features/map/overlays/TrackTypeCardOverlay.js

```diff
@@ -34,7 +34,7 @@ const TrackTypeCardOverlay = ({ map, pointerMoveEventPixel, feature }) => {
 
   useEffect(() => {
     if (overlayRef.current && overlayObjectRef.current) {
```

Comment:
That's a hell lot of ifs and nullish checks, not the goal of this PR but maybe a TODO?